### ISSUE TITLE
refactor `npm ci` so logic is in arborist instead of the cli

### ISSUE THREAD

**Issue by nlf** [npm/statusboard#476]
we have `npm ci` currently implemented by calling `loadVirtual()` then removing the `node_modules` directory, then calling `buildIdealTree` and running a cli-local validation function before finally calling `reify()` and running the relevant install scripts.

this all feels like logic that could exist as either a flag to `reify()` or a new method `cleanReify()` or something.

regardless of _how_ it gets refactored, i think this logic belongs in arborist itself and not in the `ci` command's implementation.

**Comment by ruyadorno** [comment]
Some thoughts I wanted to share on it. I digged into the `npm ci` implementation a while ago and it's really surprising that the implementation is not JUST throwing away `node_modules` and based on `loadVirtual()` only.

The way I understand it, having it run `buildIdealTree` might cause changes compared to the virtual tree which is totally not the contract of that command, also the entire logic of diffing virtual vs ideal should not apply there.

All that to say, I'd be interested in following along when that refactors takes place and also happy to help.

**Comment by nlf** [comment]
yeah, it looks like the `buildIdealTree` today is just used for validation, which shouldn't be necessary. we should be able to `loadVirtual()` and then assert that all of the edges are valid. if the edges are valid we're not going to change anything anyway, so there's no need for an ideal tree.
